addons/sale_timesheet/static/src/components/so_line_field/so_line_field.js

```diff
@@ -13,7 +13,7 @@ export class SoLineField extends Many2OneField {
             update(value, params);
             if ( // field is unset AND the old & new so_lines are different
                 !this.props.record.data.is_so_line_edited &&
-                this.props.value[0] != (value[0] && value[0].id)
+                this.props.record.data.so_line[0] != (value?.[0] && value[0]?.id)
```

Comment:
```suggestion
                this.props.record.data.so_line[0] != value[0]?.id
```
With optional chaining allowed in 16.2, you can simplify the right hand term to this.

Reply:
Hello @lvsz 
Thanks for suggestions. I have made changes. :+1: